pkg/minishift/hostfolder/hostfolder.go

```diff
@@ -0,0 +1,428 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostfolder
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/spf13/viper"
+
+	instanceState "github.com/minishift/minishift/pkg/minishift/config"
+	"github.com/minishift/minishift/pkg/minishift/util"
+)
+
+const (
+	HostfoldersAutoMountKey = "hostfolders-automount"
+)
+
+func IsAutoMount() bool {
+	return viper.GetBool(HostfoldersAutoMountKey)
+}
+
+func isHostRunning(driver drivers.Driver) bool {
+	return drivers.MachineInState(driver, state.Running)()
+}
+
+func IsHostfoldersDefined(printMessage bool) bool {
```

Comment:
both _is_ and _are_ are accepted imo. I can really live with both. What's more confusing imo again, is the bool flag. Would it not make more sense to do the logging at the caller side when the return value of _IsHostfoldersDefined_ is false?